components/director/internal/domain/application/resolver.go

```diff
@@ -374,6 +376,41 @@ func (r *Resolver) UnregisterApplication(ctx context.Context, id string) (*graph
 	return deletedApp, nil
 }
 
+// UnpairApplication missing godoc
+func (r *Resolver) UnpairApplication(ctx context.Context, id string) (*graphql.Application, error) {
+	log.C(ctx).Infof("Unpairing Application with id %s", id)
+
+	app, err := r.appSvc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	auths, err := r.sysAuthSvc.ListForObject(ctx, model.ApplicationReference, app.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.sysAuthSvc.DeleteMultipleByIDForObject(ctx, auths)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.oAuth20Svc.DeleteMultipleClientCredentials(ctx, auths)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.appSvc.Unpair(ctx, id)
```

Comment:
you are currently doing the scenario check in there - but logically if you are going to fail the whole thing because something is in formation/scenario, its better to "fail-fast" - basically do this check and fail early, rather than calling some deletes on systemauths and hydra clients first. Also DeleteMultipleClientCredentials does a rest call which we cant really rollback in case of unpair failure so right now if scenario exists unpair will fail and hydra creds will be just gone